File: src/frontends/mobile/spiza_customer/lib/models/location.dart
class Location {
  double lat;
  double lng;

  Location({required this.lat, required this.lng});

  Location.fromJson(Map<String, dynamic> json)
      : lat = json['lat'],
        lng = json['lng'];

  Location.empty()
      : lat = 0,
        lng = 0;
}

File: src/services/simulator/domain/location.go
package domain

import (
	"strconv"
	"strings"
)

type Location struct {
	Lat float64 `json:"lat"`
	Lng float64 `json:"lng"`
}

func LatLngToLocation(latlng string) Location {
	split := strings.Split(latlng, ",")
	lat, _ := strconv.ParseFloat(split[0], 64)
	lng, _ := strconv.ParseFloat(split[1], 64)
	return Location{Lat: lat, Lng: lng}
}

func (l Location) ToLatLng() string {
	return strconv.FormatFloat(l.Lat, 'f', 6, 64) + "," + strconv.FormatFloat(l.Lng, 'f', 6, 64)
}
